src/EGL/libegl.c

```diff
@@ -178,21 +182,12 @@ static EGLenum GuessPlatformType(EGLNativeDisplayType display_id)
         return EGL_PLATFORM_DEVICE_EXT;
     }
 
-    // Check if any vendor supports EGL_KHR_platform_wayland.
-    waylandSupported = EGL_FALSE;
+    // Next, try to guess what kind of display this is if we can
     glvnd_list_for_each_entry(vendor, vendorList, entry) {
-        if (vendor->supportsPlatformWayland) {
-            waylandSupported = EGL_TRUE;
-            break;
-        }
-    }
-
-    // If display_id is a dereferenceable pointer, then see if it looks like
-    // a Wayland display.
-    if (waylandSupported && _eglPointerIsDereferencable(display_id)) {
-        if (IsWaylandDisplay(display_id)) {
-            return EGL_PLATFORM_WAYLAND_KHR;
-        }
+	if (vendor->supportsPlatformWayland &&
+	    IsWaylandDisplay(display_id)) {
```

Comment:
I was going to complain about this dlopening libwayland-client.so.0 more often than necessary, but then I noticed the RTLD_NOLOAD thing, and there presumably aren't too many Wayland vendors at once anyway.